app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt

```diff
@@ -0,0 +1,113 @@
+package com.islaharper.dawnofandroid.presentation.components
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AppTopBar(
+    title: String,
+    darkTheme: Boolean,
+    dynamicTheme: Boolean,
+    onSearchClicked: () -> Unit,
+    onThemeSwitchClicked: () -> Unit,
+    onDynamicSwitchClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    var expanded by remember {
+        mutableStateOf(false)
+    }
+
+    TopAppBar(
+        title = {
+            Text(
+                text = title,
+                color = MaterialTheme.colorScheme.onPrimary
+            )
+        },
+        actions = {
+            IconButton(onClick = onSearchClicked) {
+                Icon(
+                    Icons.Default.Search,
+                    contentDescription = stringResource(R.string.search_icon),
```

Comment:
Nit:
Can we please check what TalkBack announces when we turn the screen reader on and focus on this button? 

If we look at other high quality apps, like the Google apps, I think it should say something like "Close, Button, double-tap to activate." The way this code is written, it looks like it will say "Close Icon, Button, double-tap to activate"

Reply:
Thanks I will have a look at this, I completely over looked the accessibility side.